embedchain/loaders/github.py

```diff
@@ -61,56 +65,148 @@ def _fetch_or_clone_repo(repo_url: str, local_path: str):
+                for path in file_paths:
+                    actions[url].append((action, path))
             else:
-                data = _load_file_data(file_path)
-
-            return data.get("data", [])
-
-        def _is_file_empty(file_path):
-            return os.path.getsize(file_path) == 0
+                raise ValueError(
+                    "Github loader does not support this url. Please check the documentation for the correct format - `https://docs.embedchain.ai/data-sources/github`"  # noqa: E501
+                )
+        return actions
 
-        def _is_whitelisted(file_path):
-            whitelisted_extensions = ["md", "txt", "html", "json", "py", "js", "jsx", "ts", "tsx", "mdx", "rst"]
-            _, file_extension = os.path.splitext(file_path)
-            return file_extension[1:] in whitelisted_extensions
+    def _get_github_repo_data(self, path):
+        url = f"{GITHUB_URL}/{'/'.join(path.split('/')[3:])}"
+        try:
+            with open(path, "r") as f:
+                content = f.read()
+                return [
+                    {
+                        "content": clean_string(content),
+                        "meta_data": {"url": url},
+                    }
+                ]
+        except Exception:
+            logging.info(f"Could not read file at url: {url}")
+            return None
+
+    def _get_github_api_data(self, url):
+        time.sleep(1.0)
+        response = requests.get(url, headers=self.headers)
+        response.raise_for_status()
+        if response.status_code == 200:
+            response = response.json()
+            if isinstance(response, list):
+                data = []
+                for item in response:
+                    title = item.pop("title")
+                    body = item.pop("body")
+                    format_content = f"Title: {title}\nDescription: {body}"
+                    meta_data = {"url": url}
+                    # should we add other meta data here?
+                    # meta_data.update(item)
+                    data.append(
+                        {
+                            "content": clean_string(format_content),
+                            "meta_data": meta_data,
+                        }
+                    )
+                return data
+            else:
+                title = item.pop("title")
+                body = item.pop("body")
+                format_content = f"Title: {title}\nDescription: {body}"
+                meta_data = {"url": url}
+                # should we add other meta data here?
+                # meta_data.update(item)
+                return [
+                    {
+                        "content": clean_string(format_content),
+                        "meta_data": meta_data,
+                    }
+                ]
+        else:
+            logging.info(f"[INFO] Failed to fetch data for {url}")
+            return None
+
+    def _get_github_data(self, action):
```

Comment:
Its sort of confusing right now since we have three functions here:

- _get_github_data()
- _get_github_api_data()
- _get_github_repo_data()

Can we please use more meaningful name of the function and potentially add docstring to make sure that it is understandable?